k256/src/arithmetic.rs

```diff
@@ -124,36 +133,66 @@ impl AffinePoint {
         }
     }
 
+    /// Returns a [`PublicKey`] with the SEC-1 compressed encoding of this point.
+    pub fn to_compressed_pubkey(&self) -> PublicKey {
+        PublicKey::Compressed(self.clone().into())
+    }
+
+    /// Returns a [`PublicKey`] with the SEC-1 uncompressed encoding of this point.
+    pub fn to_uncompressed_pubkey(&self) -> PublicKey {
+        PublicKey::Uncompressed(self.clone().into())
+    }
```

Comment:
I changed these to return an actual `PublicKey`, and moved their previous implementations which return a `CompressedCurvePoint` or `UncompressedCurvePoint` into the `From` impls below.

That's because I'm using `From` bounds for `FixedBaseScalarMul::Point`, but also I think having the `to_*_pubkey` methods return a `PublicKey` makes sense.